Development/nmos/video_h265.cpp

```diff
@@ -0,0 +1,100 @@
+#include "nmos/video_h265.h"
+#include "json_fields.h"
+
+namespace nmos
+{
+
+
+    // Construct additional "video/H265" parameters from the IS-04 resources
+    video_h265_parameters make_video_h265_parameters(const web::json::value& node, const web::json::value& source, const web::json::value& flow, const web::json::value& sender, const utility::string_t& sprop_parameter_sets)
+    {
+        video_h265_parameters params;
+
+        params.profile_id = nmos::fields::profile_id(flow);
+
+        params.level_id = nmos::fields::level_id(flow);
+
+        params.interop_constraints = nmos::fields::interop_constraints(flow);
+
+        return params;
+    }
+
+    // Construct SDP parameters for "video/H265", with sensible defaults for unspecified fields
+    sdp_parameters make_video_H265_sdp_parameters(const utility::string_t& session_name, const video_h265_parameters& params, uint64_t payload_type, const std::vector<utility::string_t>& media_stream_ids, const std::vector<sdp_parameters::ts_refclk_t>& ts_refclk)
+    {
+        // a=rtpmap:<payload type> <encoding name>/<clock rate>[/<encoding parameters>]
+        sdp_parameters::rtpmap_t rtpmap = { payload_type, U("H265"), 90000 };
+
+
+
+        // a=fmtp:<format> <format specific parameters>
+        // for simplicity, following the order of parameters given in VSF TR-05:2017
+        // See https://tools.ietf.org/html/rfc4566#section-6
+        sdp_parameters::fmtp_t fmtp = {
+                { sdp::fields::profile_id, utility::ostringstreamed(params.profile_id) },
+                { sdp::fields::level_id, utility::ostringstreamed(params.level_id) },
+                { sdp::fields::interop_constraints, utility::ostringstreamed(params.interop_constraints) },
```

Comment:
These all need custom functions, `utility::ostringstreamed` is not going to work - see e.g. `nmos::details::make_sdp_profile_level_id`  in the _nmos/video_h264.cpp_ file in my PR.

https://github.com/garethsb/nmos-cpp/pull/5/files#diff-ed910da386a424bbdd828d290c4f09614bc006bdfa600b916c4f39e1d5b3e020R48-R64